recipes/ucc/meta.yaml

```
@@ -0,0 +1,53 @@
+package:
+  name: ucc
+  version: v1.2.0
+
+source:
+  url: https://github.com/openucx/ucc/archive/refs/tags/v1.2.0.tar.gz
+  sha256: c1552797600835c0cf401b82dc89c4d27d5717f4fb805d41daca8e19f65e509d
+  patches:
+    - 0001-EC-CUDA-fix-build-with-12.2-and-older-arch-847.patch  # [(cuda_compiler_version or "None").startswith("12")]
+
+build:
+  skip: True  # [not linux]
+  skip: True  # [cuda_compiler_version == "None"]
+  skip: True  # [cuda_compiler_version == "11.2"]
+  number: 0
+  run_exports:
+    - {{ pin_subpackage("ucc", max_pin="x.x.x") }}
```

How did we decide on this pinning? Are we sure it needs to be this tight?

---

I was suggesting a safer card for now until we know better. https://github.com/conda-forge/staged-recipes/pull/25886#discussion_r1578819440

---

Let's ask: https://github.com/openucx/ucc/issues/968

---

Looks like the libraries have a soname. (lib/libucc.so.1.0.0). Could separate the libraries into a sonamed output.

---

Sounds like UCC is compatible within major versions ( https://github.com/openucx/ucc/issues/968#issuecomment-2077613218 )

```suggestion
  run_exports:
    - {{ pin_subpackage("ucc") }}
```